barretenberg/cpp/src/barretenberg/stdlib_circuit_builders/ultra_flavor.hpp

```diff
@@ -335,8 +335,10 @@ class UltraFlavor {
         using Base = ProvingKey_<FF, CommitmentKey>;
         using Base::Base;
 
-        ProvingKey(const size_t circuit_size, const size_t num_public_inputs)
-            : Base(circuit_size, num_public_inputs)
+        ProvingKey(const size_t circuit_size,
```

Comment:
had to make these changes to ultra and ultra_keccak because otherwise the call to the constructor was just calling the Base one directly. That meant the polynomials did not get initialized to circuit_size length leading to a segfault.

I briefly tried getting rid of the Base constructors (from the `using Base::Base;` line), but other parts of the code complained.

Reply:
actually, maybe adding default constructors to ProvingKey of each flavor would fix this and allow us to delete `using Base::Base`. I find it weird that we would want to directly call the base class constructor

Reply:
Seems ok. Calling the parent class constructor from a child class constructor seems like a really natural pattern to me

Reply:
oh, this comment wasn't referring to that pattern. It was referring to the using Base::Base line which brings in the  base class constructors to the derived class, which was causing me to hit a segfault rather than a compile error.

Like I was just calling the base class constructor when I thought it would call the derived class constructor (because I forgot to change the derived constructor to take in the commitment_key).